libs/mimir/src/utils/docker.rs

```diff
@@ -41,22 +43,24 @@ pub async fn initialize() -> Result<(), Error> {
 /// sure subsequent calls to that Elasticsearch will be successful.
 pub async fn initialize_with_param(cleanup: bool) -> Result<(), Error> {
     let mut docker = DockerWrapper::new();
-    let is_available = docker.is_container_available().await?;
-    if !is_available {
+
+    if docker.docker_config.enable && !docker.is_container_available().await? {
         docker.create_container().await?;
+
+        if !docker.is_container_available().await? {
+            return Err(Error::Misc {
+                msg: format!(
+                    "Cannot get docker {} available",
+                    docker.docker_config.container.name
+                ),
+            });
+        }
     } else if cleanup {
         docker.cleanup().await?;
```

Comment:
If docker is deactivated are you sure we should still call this function ?

Reply:
Pretty sure, the implementation of `cleanup()` bellow only connects to elasticsearch through HTTP to remove existing data. The naming is pretty bad and I'd like to change it but I deactivated my IDE tools to face my power failure :sweat: